wordpress-mobile/WordPress-Android
# Issue/16201 disable comment user data editing for registered users

Fixes #16201

This PR disables user details fields when editing comment of registered users, and enables them when editing comment of unregistered users.

I also fixed an issue where Edit option was not visible on self hosted sites in comment details.

This companion Flux-C [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2333) needs to be merged first.

To test:
- From comment details or notification open editor for a comment from registered user. 
- Confirm that the Name, Web Address and Email fields are disabled.
- Perform same check for a comment from unregistered user, and confirm that those fields are enabled and ediablte.

## Regression Notes
1. Potential unintended areas of impact
- Enabled/Disabled state of fields is wrong.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual and automated testing.

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
